language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/AddModuleToBallerinaTomlCodeActionTest.java

```diff
@@ -0,0 +1,132 @@
+import org.ballerinalang.langserver.commons.capability.InitializationOptions;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceDocumentException;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceManager;
+import org.ballerinalang.langserver.contexts.LanguageServerContextImpl;
+import org.ballerinalang.langserver.util.FileUtils;
+import org.ballerinalang.langserver.util.TestUtil;
+import org.eclipse.lsp4j.jsonrpc.Endpoint;
+import org.mockito.Mockito;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Test class to test module addition to Ballerina.toml.
+ *
+ * @since 2201.8.0
+ */
+public class AddModuleToBallerinaTomlCodeActionTest extends AbstractCodeActionTest {
+    
+    protected void setupLanguageServer(TestUtil.LanguageServerBuilder builder) {
+        builder.withInitOption(InitializationOptions.KEY_POSITIONAL_RENAME_SUPPORT, true);
+    }
+
+    @BeforeClass
+    @Override
+    public void setup() {
+        super.setup();
+
+        LanguageServerContext context = new LanguageServerContextImpl();
+        BallerinaLanguageServer languageServer = new BallerinaLanguageServer();
+        Endpoint endpoint = TestUtil.initializeLanguageSever(languageServer);
+        try {
+            Map<String, String> localProjects = Map.of("pkg1", "main.bal", "pkg2", "main.bal", "x", "main.bal", 
+                    "x.y", "main.bal");
+            List<LSPackageLoader.ModuleInfo> localPackages = getLocalPackages(localProjects,
+                    languageServer.getWorkspaceManager(), context).stream().map(LSPackageLoader.ModuleInfo::new)
+                    .collect(Collectors.toList());
+            Mockito.when(getLSPackageLoader().getLocalRepoPackages(Mockito.any())).thenReturn(localPackages);
+        } catch (Exception e) {
+            //ignore
+        } finally {
+            TestUtil.shutdownLanguageServer(endpoint);
+        }
+    }
+
+    @Test(dataProvider = "codeaction-data-provider")
+    @Override
+    public void test(String config) throws IOException, WorkspaceDocumentException {
+        super.test(config);
+    }
+
+    @DataProvider(name = "codeaction-data-provider")
+    @Override
+    public Object[][] dataProvider() {
+        return new Object[][]{
+                {"add_module1.json"},
+                {"add_module2.json"},
+                {"add_module3.json"},
+                {"add_module4.json"},
+                {"add_module5.json"},
+                {"add_module6.json"},
+                {"add_module7.json"},
+                {"add_module8.json"},
+                {"add_module9.json"},
+                {"add_module10.json"},
+        };
+    }
+
+    @Test(dataProvider = "negativeDataProvider")
```

Comment:
Shall we change this to `negative-test-data-provider` to be consistent with the names of other providers?